task/aws/resources/data_source_bucket.go

```diff
@@ -3,69 +3,60 @@ package resources
 import (
 	"context"
 	"fmt"
-	"path"
-
-	"terraform-provider-iterative/task/common"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"terraform-provider-iterative/task/common"
+	"terraform-provider-iterative/task/common/machine"
 )
 
 // NewExistingS3Bucket returns a new data source refering to a pre-allocated
 // S3 bucket.
-func NewExistingS3Bucket(client S3Client, credentials aws.Credentials, id string, region string, path string) *ExistingS3Bucket {
+func NewExistingS3Bucket(credentials aws.Credentials, storageParams common.RemoteStorage) *ExistingS3Bucket {
 	return &ExistingS3Bucket{
-		client:      client,
 		credentials: credentials,
-		region:      region,
-
-		id:   id,
-		path: path,
+		params:      storageParams,
 	}
 }
 
 // ExistingS3Bucket identifies an existing S3 bucket.
 type ExistingS3Bucket struct {
-	client      S3Client
 	credentials aws.Credentials
 
-	id     string
-	region string
-	path   string
+	params common.RemoteStorage
 }
 
 // Read verifies the specified S3 bucket is accessible.
 func (b *ExistingS3Bucket) Read(ctx context.Context) error {
-	input := s3.HeadBucketInput{
-		Bucket: aws.String(b.id),
-	}
-	if _, err := b.client.HeadBucket(ctx, &input); err != nil {
-		if errorCodeIs(err, errNotFound) {
-			return common.NotFoundError
-		}
-		return err
+	err := machine.CheckStorage(ctx, b.connection())
```

Comment:
Using rclone to verify bucket access (this will be done across all providers, sans `k8s`)